[Mehvin] iP

Mehvin opened the pull request:
# Duke
> “Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes))

Duke frees your mind of having to remember things you need to do. It's,
- ~~text~~ GUI based
- easy to learn
- ~~FAST~~ _SUPER_ FAST to use

All you need to do is,

1. download it from [here](https://github.com/Mehvin/ip/releases/download/A-Release/MEL.jar).
2. follow the [user guide](https://mehvin.github.io/ip/)
3. let it manage your tasks for you 😉

And it is **FREE**!

Features:

- [x]  Managing todos
- [x] Managing events
- [x]  Managing deadlines
- [x]  Find tasks
- [ ]  Reminders (coming soon)

If you are a Java programmer, you can use it to practice Java too. Here's the `main` method:
```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Mehvin committed:
Level 1. Greet, Echo, Exit

Mehvin committed:
Level 2. Add, List

Mehvin committed:
Level 3. Mark as Done

Mehvin committed:
Level 4. ToDos, Events, Deadlines

Mehvin committed:
Automated Text UI Testing

Test using the I/O redirection technique

Mehvin committed:
Level 5. Handle Errors

Mehvin committed:
Level 6. Delete

Mehvin committed:
Add saving tasks list feature

Mehvin committed:
Add date/time feature

Mehvin committed:
Merge branch 'branch-Level-7'

Mehvin committed:
Merge branch 'branch-Level-8'

Mehvin committed:
Fix minor date/time bugs

Mehvin committed:
Merge branch 'add-gradle-support'

Mehvin committed:
Set up gradle

Mehvin committed:
Add JUnit tests

Mehvin committed:
Divide classes into packages

Mehvin committed:
Refactor code to include Ui, TaskList classes

Mehvin committed:
Refactor code to include Parser, Storage classes

Mehvin committed:
Align to coding standard

Mehvin committed:
Add find feature

Mehvin committed:
Add JavaDoc comments

Mehvin committed:
Merge branch 'branch-Level-9'

Mehvin committed:
Merge branch 'branch-A-CodingStandard'

Mehvin committed:
Merge branch 'branch-A-JavaDoc'

Mehvin committed:
Fix minor bugs

immanuelhume commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,167 @@
+        EVENT,
+        DEADLINE
+    }
+    private static final TaskList list = new TaskList();
```
Might be better to use a more descriptive name for this variable, e.g. `tasks`.

immanuelhume commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,167 @@
+     */
+    private static void addTask(String description, TaskType taskType) {
+        switch (taskType) {
+            case TODO:
```
Based on the module's style guide, the `switch` and `case` statements should have the same indentation levels.

immanuelhume commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,89 @@
+                        case 'E':
+                            String modifiedDescription = description.split(" \\(from: ")[1];
+                            LocalDate fromDate = LocalDate.parse(modifiedDescription.split(" to: ")[0], DateTimeFormatter.ofPattern("MMM d yyyy"));
+                            LocalDate toDate = LocalDate.parse(modifiedDescription.split(" to: ")[1].split("\\)")[0], DateTimeFormatter.ofPattern("MMM d yyyy"));
```
I believe this line here exceeds the recommended length of 120 characters.

Mehvin replied:
Thanks for pointing this out!

immanuelhume commented on src/main/java/duke/task/Task.java:
```diff
@@ -0,0 +1,58 @@
+     *
+     * @param status Status of mark task.
+     */
+    public void setStatus(boolean status) {
```
Perhaps `setStatus` is not the best name here, would `setDone` be better?

Mehvin replied:
Great idea! Thanks!

immanuelhume reviewed (commented):
Good work, I liked that all your JavaDocs were in place.

Mehvin committed:
Resolve peer review feedbacks

Mehvin committed:
Add Checkstyle support

Mehvin committed:
Add GUI

Mehvin committed:
Merge branch 'branch-Level-10'

Mehvin committed:
Fix minor GUI bug

Mehvin committed:
Add assertions

Add assertions

Assertions are used to define assumptions about the program state so that the runtime can verify them.

Asserts are added mainly whenever users perform an action that requires change to a task.

This was done to ensure that the modification of the task is done correctly and no error occurs.

Mehvin committed:
Improve code quality

- Magic numbers and strings are converted to named constant
- Refactor code to improve readability and simplicity (KISS)

Mehvin committed:
Merge pull request #1 from Mehvin/branch-A-Assertions

Add assertions

Mehvin committed:
Merge branch 'master' into branch-A-CodeQuality

Mehvin committed:
Merge pull request #2 from Mehvin/branch-A-CodeQuality

Improve code quality

Mehvin committed:
Set up CI

GitHub Actions is a Continuous Integration (CI) tool integrated into GitHub.

GitHub Actions will run the workflow (as per the .yml file) every time certain project events are triggered.

Mehvin committed:
Add help and sort features

C-Help
Users can get a list of valid commands by typing 'help'.
Additionally, by typing 'help [command]' users will get in-depth information about the specific command (e.g. format and function of command)

C-Sort
Sorting items will be done every time a new task is added.
Tasks are sorted in priority order (From highest to lowest priority - deadlines, events, todos).
For similar types of task, they are sorted in alphabetical order based on the description of the task.

Mehvin committed:
Merge branch 'branch-BCD-Extension'

Mehvin committed:
Add bye command into help

Mehvin committed:
Improve GUI

Improve GUI for user experience and added unique personality.
Changed display format and UI related to javaFX by adding logo, title and dialog bubble.

Mehvin committed:
Add more error handling

To handle user input error

Mehvin committed:
Release user guide

Product website: https://mehvin.github.io/ip/

Mehvin committed:
Fix minor bugs to build jar file

Mehvin committed:
Update README

To include links for users to navigate to downloads

Mehvin committed:
Add auto-scroll feature

Mehvin committed:
Update README.md
